sdk/scheduler/src/main/java/org/apache/mesos/offer/constrain/PlacementRule.java

```diff
@@ -10,10 +14,12 @@
  *
  * Accepts an Offer and returns a copy of that Offer with any disallowed Resources filtered out, or
  * may return the original Offer without copying if everything passes the filter. If the entire
- * Offer is rejected, the return value will be a non-{@code null} Offer with zero Resources. A
- * PlacementRule may contain multiple internal checks, and may even be composed of other sub-Rules.
+ * Offer is rejected, the return value will be a non-{@code null} Offer with zero Resources.
+ *
+ * PlacementRules may contain multiple internal checks, and may even be a composition of other
+ * PlacementRules.
  */
-@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS)
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME)
```

Comment:
What's the consequence of this?  No writing rules that have the same class name in different packages?